### ISSUE TITLE
Uninformative error message in `cabal sdist`

### ISSUE THREAD

**Issue by fendor** [haskell/cabal#10615]
**Describe the bug**
When `cabal sdist` encounters a path that is too long, it errors out with the error message:

```
> cabal sdist
Error: [Cabal-7039]
Error packing sdist: File name too long
```

Which is not useful to understand what went wrong.

**To Reproduce**

I wrote a small reproducer in https://github.com/fendor/cabal-sdist-tar-unclear-error

```sh
$ git clone git@github.com:fendor/cabal-sdist-tar-unclear-error.git
$ bash reproducer.sh
```

The reproducer is slightly weird, as this is what I encountered when running `cabal sdist` on the HLS codebase, where I apparently ran `cabal build` before in a test project.
The globs 
```
extra-source-files:
    test/data/**/*.hs
    test/data/**/*.cabal
```

are an antipattern, and pick up the `Path_*.hs` file in `dist-newstyle` which is a fairly long name.

I identified the issue, by logging the file which failed:
```
Error packing sdist "cabal-sdist-0.1.0.0/test/data/really-long-name-to-trigger-something/dist-newstyle/build/x86_64-linux/ghc-9.6.6/really-long-name-to-trigger-something-0.1.0.0/build/autogen/PackageInfo_really_long_name_to_trigger_something.hs": File name too long
```

I am not sure which Path length issue I am encountering here, presumably `tar`?

Everything works well, if I delete `dist-newstyle` in the test project before running `cabal sdist`.

**Expected behavior**
Should not crash or give a comprehensive error message

**System information**
 - `Linux hugin 6.12.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 22 Nov 2024 16:04:27 +0000 x86_64 GNU/Linux`
 - `cabal 3.12.1`, `ghc 9.6.6

**Additional context**

I don't think the behaviour of cabal when building my reproducer is wrong. The error message is just not helpful.

**Comment by Mikolaj** [comment]
Is the fix relatively obvious? If not, let's design the fix here and if it's not a one-liner, but not soul-killing, maybe let's add a "newcomer" label?

**Comment by ffaf1** [comment]
After a bit of investigating, the error if from `tar` itself.

So I searched `tar` repository to see if someone took care of adding a more informative message.

And someone did!

Imagine the shock when I found out [it was me](https://github.com/haskell/tar/pull/76).

Sorry for having had you write this very informative issue+reproduce @fendor, I totally forgot about it.

Duplicate:  #9438

If this fits your bill, we can close this Fendor.
